src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/TimerTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices.JavaScript;
+using Xunit;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Private.Runtime.InteropServices.JavaScript.Tests.System.Runtime.InteropServices.JavaScript
+{
+    public class TimerTests : IDisposable
+    {
+        // , stack: ${(new Error()).stack}
+        Function _wrapJsSetTimer = new Function(@"
+            globalThis.callCounter = 0;
+            console.log(`reset callCounter: ${globalThis.callCounter}`)
+            globalThis.originalSetTimeout = globalThis.setTimeout;
+            globalThis.setTimeout = (cb,time) => {
+                globalThis.callCounter++;
+                console.log(`callCounter: ${globalThis.callCounter} now:${Date.now()} delay:${time} cb:${cb.name}`)
+                globalThis.originalSetTimeout(cb, time);
+            };
+            ");
+
+        Function _getCallCount = new Function(@"
+            return globalThis.callCounter;
+            ");
+
+        Function _cleanupJsSetTimer = new Function(@"
+            console.log(`cleanup setTimeout, callCounter: ${globalThis.callCounter}`)
+            globalThis.setTimeout = globalThis.originalSetTimeout;
+            ");
+
+        public TimerTests()
+        {
+            _wrapJsSetTimer.Call();
+        }
+
+
+        public void Dispose()
+        {
+            _cleanupJsSetTimer.Call();
+        }
+
+        [Trait("Category", "Pavel")]
+        [Fact]
+        public async Task NoTimer()
+        {
+            await Task.Delay(50);// make sure timers from other tests are gone
```

Comment:
it would be good to verify this robustly, maybe by tracking a count of the # of registered timers that haven't fired yet. just if you plan to merge these tests later (I think it's worth merging them)

also as-is this test may fail since the order individual tests run in is randomized iirc

Reply:
Yes, our XHarness thread-less runner makes it impossible to test reliably. See https://github.com/dotnet/xharness/blob/2863beb74c1abad7ddfe9ef7afea6fc2605e8a03/src/Microsoft.DotNet.XHarness.TestRunners.Xunit/ThreadlessXunitTestRunner.cs#L68

I will convert it to "Sample" like code which doesn't use XHarness.

Reply:
@maraf improved that !
https://github.com/dotnet/xharness/pull/790 

Would it be enough to get rid of Timers in the environment ?